File: herencia/coworking-csharp/Course.OOP.Inheritance.Business/Media/Speaker.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Course.OOP.Inheritance.Business.Media
{
    public class Speaker: ISchedulable
    {

        private double Weight { get; set; }
        private bool IsBluetooth { get; set; }

        public Speaker(double weight, bool isBluetooth)
        {
            this.Weight = weight;
            this.IsBluetooth = isBluetooth;
        }

        public bool IsAvailable(DateTime start, DateTime end)
        {
            //Implementar lógica para definir disponibilidad.
            return true;
        }

    }
}

File: herencia/coworking-java/src/main/java/course/oop/inheritance/media/Speaker.java
package course.oop.inheritance.media;

import course.oop.inheritance.Schedulable;

import java.time.LocalDateTime;

public class Speaker implements Schedulable {

    private double weight;
    private boolean isBluetooth;

    public Speaker(double weight, boolean isBluetooth) {
        this.weight = weight;
        this.isBluetooth = isBluetooth;
    }

    public boolean isAvailable(LocalDateTime start, LocalDateTime end) {
        //Implementar lógica para definir disponibilidad.
        return true;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public boolean isBluetooth() {
        return isBluetooth;
    }

    public void setBluetooth(boolean bluetooth) {
        isBluetooth = bluetooth;
    }


}
